pkg/util/util_test.go

```diff
@@ -0,0 +1,84 @@
+package util
+
+import (
+	"encoding/hex"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	testCA1Subj = "301e311c301a060355040313136f617574682d70726f78792074657374206361"
+	testCA1     = `-----BEGIN CERTIFICATE-----
```

Comment:
I wonder if there's a way we can generate certificates in memory, easily, during the testing so that we don't have to commit certificates into the tests.

Out of interest, how did you create this certificate?

Reply:
If I were to generate the certificate, this would bring a heap of crypto/x509 and encoding/pem noise to the test.  I think this might be worse than the inline string.   I'm happy to take the generation route if preferred.  The implementation of the test and the test data came from: https://github.com/openshift/oauth-proxy/commit/b3380fc94a13616549cede5e3841d09d371473f3.    The original GetCertPool implementation came from the same project.

Reply:
I would love to have a comment that explains how these were made, or at least, how we could regenerate some if we ever wanted to change them, don't suppose you happen to know?

Reply:
I've added the OpenSsl command used to generate the certificate test data to test.  I also changed the test to assert the contents of the pool based in subject names (textual) which tallies with the `-subj` command.